R-package/tests/testthat/test_dataset.R

```diff
@@ -218,6 +218,10 @@ test_that("lgb.Dataset: Dataset should be able to construct from matrix and retu
 })
 
 test_that("cpp errors should be raised as proper R errors", {
+  testthat::skip_if(
+    Sys.getenv("COMPILER", "") == "MSVC" && as.integer(R.Version()[["major"]]) < 4L
+    , message = "Skipping on R 3.x and Visual Studio 16 2019"
```

Comment:
```suggestion
    , message = "Skipping on R 3.x and Visual Studio"
```

Reply:
ah good point, thanks. Updated in https://github.com/microsoft/LightGBM/pull/5448/commits/121e117edeb734e320cc85f1bd0a6660557823ef.

(I chose not to do this in the browser so that this change + the one for https://github.com/microsoft/LightGBM/pull/5448#discussion_r967875358 only generated one new commit, to save CI time).